homeassistant/components/reolink/host.py

```diff
@@ -362,57 +362,49 @@ def unregister_webhook(self):
     async def handle_webhook(
         self, hass: HomeAssistant, webhook_id: str, request: Request
     ) -> None:
-        """Shield the incoming webhook callback from cancellation."""
-        shielded_future = asyncio.shield(
-            self._handle_webhook(hass, webhook_id, request)
-        )
+        """Read the incoming webhook from Reolink for inbound messages and schedule processing."""
         _LOGGER.debug("Webhook '%s' called", webhook_id)
-        if not self._webhook_reachable.is_set():
-            self._webhook_reachable.set()
-            ir.async_delete_issue(self._hass, DOMAIN, "webhook_url")
-        await shielded_future
-
-    async def _handle_webhook(
-        self, hass: HomeAssistant, webhook_id: str, request: Request
-    ) -> None:
-        """Handle incoming webhook from Reolink for inbound messages and calls."""
         try:
             data = await request.text()
-        except ConnectionResetError:
-            # We lost the connection before reading the message, fallback to polling
-            # No need for a background task here as we already know the connection is lost
+            if not data:
+                _LOGGER.debug(
+                    "Webhook '%s' triggered with unknown payload: %s", webhook_id, data
+                )
+        except (asyncio.CancelledError, ConnectionResetError):
```

Comment:
I think you want to swallow `ConnectionResetError` here since its going to be handled by your finally.  I'd split this block and not raise on `ConnectionResetError` but still raise on `CancelledError` so the cancellation propogates

I'd also set data to `None` above so you don't have to set it in the except

`data: str | None = None`

Reply:
Done, thanks for the suggestions!